x/inflation/keeper/integration_test.go

```diff
@@ -1,70 +1,181 @@
-				expected := provision.Mul(distribution)
+						s.CommitAfter(time.Second * 6) // commit next block
+					})
+					It("should increase the epoch number ", func() {
+						epochInfo, _ := s.app.EpochsKeeper.GetEpochInfo(s.ctx, "day")
+						Expect(epochInfo.CurrentEpoch).To(Equal(epochNumber + 1))
+					})
+					It("should not increase the skipped epochs number", func() {
+						skippedAfter := s.app.InflationKeeper.GetSkippedEpochs(s.ctx)
+						Expect(skippedAfter).To(Equal(skipped + 1))
+					})
+				})
+
+				When("the epoch start time has caught up with the block time", func() {
+					BeforeEach(func() {
+						// commit next 4 blocks to trigger afterEpochEnd hook several times
+						// and let EpochStartTime catch up with BlockTime
+						s.CommitAfter(time.Second * 6)
+						s.CommitAfter(time.Second * 6)
+						s.CommitAfter(time.Second * 6)
+						s.CommitAfter(time.Second * 6)
+
+						epochInfo, found := s.app.EpochsKeeper.GetEpochInfo(s.ctx, "day")
+						s.Require().True(found)
+						epochNumber = epochInfo.CurrentEpoch
+
+						skipped = s.app.InflationKeeper.GetSkippedEpochs(s.ctx)
+
+						s.CommitAfter(time.Second * 6) // commit next block
+					})
+					It("should not increase the epoch number ", func() {
+						epochInfo, _ := s.app.EpochsKeeper.GetEpochInfo(s.ctx, "day")
+						Expect(epochInfo.CurrentEpoch).To(Equal(epochNumber))
+					})
+					It("should not increase the skipped epochs number", func() {
+						skippedAfter := s.app.InflationKeeper.GetSkippedEpochs(s.ctx)
+						Expect(skippedAfter).To(Equal(skipped))
+					})
+
+					When("epoch number passes epochsPerPeriod + skippedEpochs and inflation re-enabled", func() {
+						BeforeEach(func() {
+							params := s.app.InflationKeeper.GetParams(s.ctx)
+							params.EnableInflation = true
+							s.app.InflationKeeper.SetParams(s.ctx, params)
+
+							epochInfo, _ := s.app.EpochsKeeper.GetEpochInfo(s.ctx, "day")
+							epochNumber := epochInfo.CurrentEpoch // 6
+
+							epochsPerPeriod := int64(1)
+							s.app.InflationKeeper.SetEpochsPerPeriod(s.ctx, epochsPerPeriod)
+							skipped := s.app.InflationKeeper.GetSkippedEpochs(s.ctx)
+							s.Require().Equal(epochNumber, epochsPerPeriod+int64(skipped))
+
+							provision, found = s.app.InflationKeeper.GetEpochMintProvision(s.ctx)
+							s.Require().True(found)
+							fmt.Println(provision)
+
+							s.CommitAfter(time.Hour * 23) // commit before next full epoch
+							provisionAfter, _ := s.app.InflationKeeper.GetEpochMintProvision(s.ctx)
+							s.Require().Equal(provisionAfter, provision)
+
+							s.CommitAfter(time.Hour * 2) // commit after next full epoch
+						})
+
+						It("should recalculate the EpochMintProvision", func() {
+							provisionAfter, _ := s.app.InflationKeeper.GetEpochMintProvision(s.ctx)
 
-				Expect(balanceCommunityPool.IsZero()).ToNot(BeTrue())
-				Expect(balanceCommunityPool.AmountOf(denomMint).GT(expected)).To(BeTrue())
+							Expect(provision).ToNot(Equal(provisionAfter))
```

Comment:
should we check for equality here?

Reply:
@fedekunze  I expect that the provision is recalculated, so I'm making sure that they MUST NOT be equal. Could you explain w
hat you mean?

Reply:
I meant that we should also check that the value matches the new provision. So basically create an Equal with the new hardcoded provision value, so that we ensure that any involuntary changes to the inflation break the test